### PR TITLE
Feat: Created School Creation Api

### PR DESCRIPTION
**Pull request by pratik-edu**

### CODE REVIEW

File: src/controllers/school.controller.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// controllers/SchoolController.ts
+import { Request, Response } from 'express';
+import SchoolService from '@/services/school.service';
+import { createSchoolRequestBody } from './typings/school.controller';
```
Comment (HiAkshat) [inline src/controllers/school.controller.ts:4]: Write import comments

File: src/controllers/school.controller.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * ````
+ * POST /api/v1/platform/school/create
+ * ````
+ * @param req - The HTTP request object containing the school details.
+ * @param res - The HTTP response object used to send the response back to the client.
+ */
+class SchoolController {
+  private schoolService = new SchoolService();
+
+  public createSchool = async (req: Request<{}, {}, createSchoolRequestBody>, res: Response) => {
```
Comment (HiAkshat) [inline src/controllers/school.controller.ts:19]: Write function comment

File: src/dao/school.dao.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import School, { ISchool } from '@/models/school.model';
+import { ISchoolDetails } from '@/typings/school';
```
Comment (HiAkshat) [inline src/dao/school.dao.ts:2]: Add import comments

File: src/formatters/authenticate.formatter.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 // Typings
 import { CreateUserInDb, UserSignupData } from '@/typings/authenticate';
+import { generateUniqueUserId } from '@/utils/auth.utils';
```
Comment (HiAkshat) [inline src/formatters/authenticate.formatter.ts:3]: Add import comments

File: src/services/school.service.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+// services/SchoolService.ts
+import SchoolDAO from '@/dao/school.dao';
+import { ISchoolDetails } from '@/typings/school';
```
Comment (HiAkshat) [inline src/services/school.service.ts:3]: Add import comments

**Review by HiAkshat — APPROVED** [review]

**Review by tusharthakurepc1 — APPROVED** [review]

**Pull request merged**